Appium Quickstart (#554)
saucelabs/sauce-docs: docs/mobile-apps/automated-testing/appium.md

### Step 2: Get Test Code
Open and review your test code in the [Sauce Appium sample repo](https://github.com/saucelabs-training/quickstart-appium-java/tree/main/src/test/java/tests), and open the test file located in `/src/test/java/tests/EmuSim` directory called `AppiumAndroidEmuWebTest.java`.

Update the `testngXmlFile` tag [in `pom.xml`](https://github.com/saucelabs-training/quickstart-appium-java/blob/main/pom.xml) to use the correct TestNG xml file.

```
<testngXmlFile>appium_android_emu_web_test.xml</testngXmlFile>
```

You can see the contents of this file in the [`/resources/xml` directory](https://github.com/saucelabs-training/quickstart-appium-java/tree/main/src/test/resources/config).

### Step 3: Set Up Capabilities
[Update your Capabilities](https://saucelabs.com/platform/platform-configurator) to run on Sauce Labs Emulators to have these required Appium Capabilities to run an Android web browser test:

```
deviceName: “Android GoogleAPI Emulator”
platforName: “Android”
automationName: “UiAutomator2”
```

### Step 4: Add Capabilities for the Browser

> Depending on which browser you would like to test on, you need to [set the capabilities](https://docs.saucelabs.com/dev/test-configuration-options#desktop-browser-capabilities-sauce-specific--optional) to test on that browser:

Reviewer: ```suggestion
[Set the capabilities](https://docs.saucelabs.com/dev/test-configuration-options#desktop-browser-capabilities-sauce-specific--optional) according to the browser on which you want to test:
```

```
browserName: "Chrome"
```
At a minimum, you must set the `browserName` capability, however different browsers have other options you can set.

<img src={useBaseUrl('img/mobile-apps/appium-web-capabilities.png')} alt="Appium Capabilities" width="800"/>

> ## Test a Mobile App on Sauce Labs Real Devices

Reviewer 2: Why are we talking about `Sauce Labs Real Devices` if the intro only mentions EMUSIM?

Author: Updated the intro to this:

This page is intended to provide you with instructions on how to quickly set up Appium tests with Java using the TestNG test runner on an Android Emulator or an iOS Real Deives, and run tests using the Sauce Labs App with the Virtual Device and Real Device Cloud. To see a comprehensive list of test examples, [check out the appium directory in our demo-java repo](https://github.com/saucelabs-training/demo-java/tree/dff5fd61b8e152efe59e4a8c9e75c644de4e51e0/appium-examples).

```

You can view the contents of the .xml file in the`/resources/xml` [directory](https://github.com/saucelabs-training/quickstart-appium-java/tree/main/src/test/resources/config).

### Step 3: Set Up Your App
Use your own app, or grab the latest [Swag Labs App](https://github.com/saucelabs/sample-app-mobile/releases).

On the [Sauce Labs App](https://app.saucelabs.com/live/app-testing), you can upload an .apk, .ipa, or .zip file, or use the API for [Mobile Application Storage](https://docs.saucelabs.com/mobile-apps/app-storage/index.html#uploading-apps-via-rest-api).

The easiest way to do this is to set environment variables for your Sauce username & access key, then use this command:

```
curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
--request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' \
--form 'payload=@"/Users/lindsaywalker/Git/Example_Tests/iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"' \
--form 'name="iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"'
```

### Step 4: Set Up Capabilities
Update your Capabilities to run on Sauce Labs Emulators to have the minimum required Appium Capabilities to run an iOS App test:

```

> String appName = "iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa";

Reviewer 2: - An Android emulator or real device needs to have an `.apk`-file
- An iOS simulator needs to have an `.app` file which needs to be zipped for the Sauce Labs amulators
- An iOS real device needs to have an `.ipa`-file

The above section is messing the three mentioned points above up into

`Update your Capabilities to run on Sauce Labs Emulators to have the minimum required Appium Capabilities to run an iOS App test`

with a cap like this

`iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa` 

Can we explain this properly in the docs and also provide a correct example?

Author: Updated step 3 to only say .ipa, and added a note about diff file types:

### Step 3: Set Up Your App
Use your own app, or grab the latest [Swag Labs App](https://github.com/saucelabs/sample-app-mobile/releases).

On the [Sauce Labs App](https://app.saucelabs.com/live/app-testing), you can upload an .ipa file, or use the API for [Mobile Application Storage](https://docs.saucelabs.com/mobile-apps/app-storage/index.html#uploading-apps-via-rest-api).

The easiest way to do this is to set environment variables for your Sauce username & access key, then use this command:

```
curl -u "$SAUCE_USERNAME:$SAUCE_ACCESS_KEY" --location \
--request POST 'https://api.us-west-1.saucelabs.com/v1/storage/upload' \
--form 'payload=@"/Users/lindsaywalker/Git/Example_Tests/iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"' \
--form 'name="iOS.RealDevice.SauceLabs.Mobile.Sample.app.2.7.1.ipa"'
```
:::note
Different file types are required for testing in different environments:
* An Android emulator or real device needs to have an .apk-file
* An iOS simulator needs to have an .app file which needs to be zipped for the Sauce Labs emulators (.zip file upload)
* An iOS real device needs to have an .ipa-file
:::

```

```
platformName: "iOS"

> deviceName: "iPhone 12"

Reviewer 2: This is an example for a Real device (simulators are always post fixed with ` Simulator`. Can we be consistent with our examples?

Author: My mistake- this isn't an EMU test, it's an RDC, I will correct that typo, should say 'Sauce Labs Real Devices' instead of 'Sauce Labs Emulators' in the paragraph at the top of the section, will do a commit to update.

Is this a good way to write an RDC cap? "iPhone 12" ? or should I add the word "Device" at the end?

## Run Your Tests

From you machine, either use the command `mvn clean test` or right click on the test you would like to run from your IDE to run your tests.  If you would like to skip Step 2, simply add the flag `-DtestngXmlFile=appium_ios_rdc_app_test.xml` and to change data centers, add the flag `-Dregion=us`

:::note
Java tests using the TestNG framework rely on the `pom.xml` file to import dependencies. Make sure the dependencies like [Appium Java Binding version](https://github.com/appium/java-client) and [TestNG version](https://testng.org/doc/download.html) are all up to date in the properties tag of `pom.xml`
```
<properties>
    <testngXmlDir>src/test/resources/config</testngXmlDir>
    <testngXmlFile>appium_ios_rdc_app_test.xml</testngXmlFile>
    <testng.version>6.14.3</testng.version>
    <commons-logging.version>1.2</commons-logging.version>
    <selenium.version>3.141.59</selenium.version>
    <maven-failsafe-plugin.version>2.19.1</maven-failsafe-plugin.version>
    <appium.version>7.5.1</appium.version>
</properties>
```
:::

### Optional Appium Capabilities

> Sauce Labs will set default values for platform (operating system) and Appium versions, however you may want to specify versions you know are supported and compatible to aviod errors.

Reviewer 2: `aviod` needs to be `avoid`

Author: done

---
id: appium

> title: Quickstart

Reviewer 3: This needs to be 'Appium Quickstart'
